run_kraken.py

```diff
@@ -12,6 +12,88 @@
+                    logging.info("Cerberus status is not healthy and post action scenarios "
+                                 "are still failing")
+        else:
+            logging.info("Received a go signal from Ceberus, the cluster is "
+                         "healthy. Test passed.")
+
+            if failed_post_scenarios:
+                if config['kraken']['exit_on_failure']:
+                    logging.info("Cerberus status is healthy but post action scenarios "
+                                 "are still failing, exiting kraken run")
+                    sys.exit(1)
+                else:
+                    logging.info("Cerberus status is healthy but post action scenarios "
+                                 "are still failing")
+    else:
+        if config['kraken']['exit_on_failure']:
+            if failed_post_scenarios:
+                logging.info("Post action scenarios are failing, exiting kraken run")
+                sys.exit(1)
+
+
+# Perform the post scenario actions to see if components recovered
+def post_actions(kubeconfig_path, scenario, failed_post_scenarios):
+    for failed_scenario in failed_post_scenarios:
+        post_action_output = runcommand.invoke("powerfulseal autonomous "
+                                               "--use-pod-delete-instead-of-ssh-kill"
+                                               " --policy-file %s --kubeconfig %s --no-cloud"
+                                               " --inventory-kubernetes --headless"
+                                               % (failed_scenario, kubeconfig_path))
+        if "ERROR" not in post_action_output:
+            failed_post_scenarios.remove(failed_scenario)
+        else:
+            logging.info('Post action scenario ' + str(failed_scenario) + "is still failing")
+            error_message = post_action_output.split("ERROR")[1].split('\n')[0]
+            logging.info("Error message was " + str(error_message))
+    # check post actions
+    if len(scenario) > 1:
+        if scenario[1].endswith(".yaml") or scenario[1].endswith(".yml"):
+            post_action_output = runcommand.invoke("powerfulseal autonomous "
+                                                   "--use-pod-delete-instead-of-ssh-kill"
+                                                   " --policy-file %s --kubeconfig %s --no-cloud"
+                                                   " --inventory-kubernetes --headless"
+                                                   % (scenario[1], kubeconfig_path))
+            # read output to make sure no error
+            if "ERROR" in post_action_output:
+                error_message = post_action_output.split("ERROR")[1].split('\n')[0]
+                logging.info("Error message was " + str(error_message))
+                failed_post_scenarios.append(scenario[1])
+            else:
+                logging.info(scenario[1] + " post action checks passed")
+        else:
+            # invoke custom bash script
```

Comment:
I think it would be nice to provide the user with an option to provide post actions using a Python file in addition to bash script as this project is based on Python and the user might be familiar with Python over shell script. Thoughts?

Reply:
Yes I agree, I can add in a check to see if the given post action scenario is a python file and run accordingly.